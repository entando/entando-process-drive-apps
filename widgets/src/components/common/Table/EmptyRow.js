import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';
import Typography from '@material-ui/core/Typography';

const styles = {
  root: {
    background: 'transparent !important',
    '& td': {
      textAlign: 'center',
      padding: 0,
    },
  },
  text: {
    color: '#ccc',
  },
};

const EmptyRow = ({ classes, colspan, text, height }) => (
  <tbody>
    <tr className={classes.root}>
      <td colSpan={colspan} style={{ height }}>
        <Typography variant="body1" className={classes.text}>
          {text}
        </Typography>
      </td>
    </tr>
  </tbody>
);

EmptyRow.propTypes = {
  classes: PropTypes.shape({
    root: PropTypes.string,
    text: PropTypes.string,
  }),
  colspan: PropTypes.number,
  text: PropTypes.string,
  height: PropTypes.number,
};

EmptyRow.defaultProps = {
  classes: {},
  colspan: 1,
  text: 'No data to display',
  height: 250,
};

export default withStyles(styles)(EmptyRow);
