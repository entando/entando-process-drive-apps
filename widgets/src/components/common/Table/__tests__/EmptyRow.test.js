import React from 'react';
import { render } from '@testing-library/react';
import EmptyRow from 'components/common/Table/EmptyRow';

describe('<EmptyRow />', () => {
  it('renders snapshot correctly', () => {
    const { container } = render(
      <table>
        <EmptyRow />
      </table>
    );

    expect(container).toMatchSnapshot();
  });

  it('renders snapshot correctly with custom message', () => {
    const { container } = render(
      <table>
        <EmptyRow text="test message" />
      </table>
    );

    expect(container).toMatchSnapshot();
  });
});
