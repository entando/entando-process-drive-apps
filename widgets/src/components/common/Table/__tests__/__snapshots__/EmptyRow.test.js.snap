// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyRow /> renders snapshot correctly 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="EmptyRow-root-1"
      >
        <td
          colspan="1"
          style="height: 250px;"
        >
          <p
            class="MuiTypography-root EmptyRow-text-2 MuiTypography-body1"
          >
            No data to display
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<EmptyRow /> renders snapshot correctly with custom message 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="EmptyRow-root-33"
      >
        <td
          colspan="1"
          style="height: 250px;"
        >
          <p
            class="MuiTypography-root EmptyRow-text-34 MuiTypography-body1"
          >
            test message
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
